packages/block-library/src/spacer/controls.js

```diff
@@ -57,19 +56,18 @@ function DimensionInput( { label, onChange, isResizing, value = '' } ) {
 	return (
 		<>
 			{ ( ! spacingSizes || spacingSizes?.length === 0 ) && (
-				<BaseControl label={ label } id={ inputId }>
-					<UnitControl
-						id={ inputId }
-						isResetValueOnUnitChange
-						min={ MIN_SPACER_SIZE }
-						onChange={ handleOnChange }
-						style={ { maxWidth: 80 } }
-						value={ computedValue }
-						units={ units }
-					/>
-				</BaseControl>
+				<UnitControl
+					id={ inputId }
+					isResetValueOnUnitChange
+					min={ MIN_SPACER_SIZE }
+					onChange={ handleOnChange }
+					style={ { maxWidth: '50%' } }
```

Comment:
> The `50%` width still feels arbitrary. I wonder if it should have the same width as it has when it has the slider control.

I considered that. Alas, that width is something like 104.some decimals pixels because it's determined by the flax layout. Not a real number we can use. I'd agree 50% is still arbitrary... but it is a little better 😄

Reply:
Aside consideration: this is the kind of problems to face when the design asks for something the base components aren't designed for. Local overrides, custom, ad-hoc implementation should generally be avoided.

Reply:
If we want to place it on a standard 2-column controls grid, I would recommend something like this:

```diff
diff --git a/packages/block-library/src/spacer/controls.js b/packages/block-library/src/spacer/controls.js
index 97e47e4d15..7d09af9969 100644
--- a/packages/block-library/src/spacer/controls.js
+++ b/packages/block-library/src/spacer/controls.js
@@ -56,17 +56,18 @@ function DimensionInput( { label, onChange, isResizing, value = '' } ) {
 	return (
 		<>
 			{ ( ! spacingSizes || spacingSizes?.length === 0 ) && (
-				<UnitControl
-					id={ inputId }
-					isResetValueOnUnitChange
-					min={ MIN_SPACER_SIZE }
-					onChange={ handleOnChange }
-					style={ { maxWidth: '50%' } }
-					value={ computedValue }
-					units={ units }
-					label={ label }
-					__next40pxDefaultSize
-				/>
+				<div className="block-library-spacer__controls-grid">
+					<UnitControl
+						id={ inputId }
+						isResetValueOnUnitChange
+						min={ MIN_SPACER_SIZE }
+						onChange={ handleOnChange }
+						value={ computedValue }
+						units={ units }
+						label={ label }
+						__next40pxDefaultSize
+					/>
+				</div>
 			) }
 			{ spacingSizes?.length > 0 && (
 				<View className="tools-panel-item-spacing">
diff --git a/packages/block-library/src/spacer/editor.scss b/packages/block-library/src/spacer/editor.scss
index e0c700795c..f2ca2ef985 100644
--- a/packages/block-library/src/spacer/editor.scss
+++ b/packages/block-library/src/spacer/editor.scss
@@ -46,3 +46,9 @@
 		height: 100% !important;
 	}
 }
+
+.block-library-spacer__controls-grid {
+	display: grid;
+	grid-template-columns: repeat(2, 1fr);
+	gap: 16px;
+}
```

(Ideally I want some kind of standard component to handle this layout #43423)

However, seeing that [the other UnitControl](https://private-user-images.githubusercontent.com/1682452/350939995-5cb69420-6105-4b6f-aedd-ef2565b1913e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4Mzg5MDMsIm5iZiI6MTcyMTgzODYwMywicGF0aCI6Ii8xNjgyNDUyLzM1MDkzOTk5NS01Y2I2OTQyMC02MTA1LTRiNmYtYWVkZC1lZjI1NjViMTkxM2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcyNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MjRUMTYzMDAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YTMwNzZmMDkzMjQ2N2RkYjYzZDdiNDJiNDI2YmM5ZjkyYzZmMzhhNjMxMTI5MTIxYjViZGNmOTYzMWYwNDI5NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.7iDkIhf8F_e01rM2tBtitaBtFwMlIs9Pg2LwClRW3fk) is full-width, I guess this one could just be full-width as well.

Reply:
> I want some kind of standard component to handle this l

Yes that would be better. I can try your CSS workaround but that's in some way an ad-hoc, custom override that should be avoided.

> seeing that the other UnitControl

The linked image gives a 404 error.

Reply:
Oops, I was trying to link to this screenshot that was in your original issue:

![The other UnitControl](https://github.com/user-attachments/assets/5cb69420-6105-4b6f-aedd-ef2565b1913e)